File: factors.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Expected 1 argument, got %d\n", argc - 1);
        return 1;
    }

    char *file_path = argv[1];
    FILE *file = fopen(file_path, "r");

    if (file == NULL) {
        perror("Error opening file");
        return 1;
    }

    char line[100];
    while (fgets(line, sizeof(line), file) != NULL) {
        long long number = atoll(line);
        for (long long i = 2; i < number; ++i) {
            if (number % i == 0) {
                printf("%lld=%lld*%lld\n", number, number / i, i);
                break;
            }
        }
    }

    fclose(file);
    return 0;
}

File: factors.py
#!/usr/bin/python3

import time
start_time = time.time()
# -----------------------------------------------------------------------------
import sys

if len(sys.argv) != 2:
    print('expected 1 argument got 0')
else:
    # Open the file in read mode
    file_path = sys.argv[1]
    with open(file_path, 'r') as file:
        # Read each line in the file
        for line in file:
            # Process each line as needed
            a = line.strip()
            number = int(a)
            for i in range(2, int(line)):
                if number % i == 0:
                    ans = i
                    print(f"{number}={int(number/i)}*{i}")
                    break
# -----------------------------------------------------------------------------
end_time = time.time()

elapsed_time = end_time - start_time
print(f"Time taken: {elapsed_time:.4f} seconds")
